src/query/service/src/interpreters/interpreter_copy.rs

```diff
@@ -483,3 +484,179 @@ fn fill_const_columns(
     })?;
     Ok(())
 }
+
+#[allow(clippy::too_many_arguments)]
+pub fn append_data_and_set_finish(
+    main_pipeline: &mut Pipeline,
+    source_schema: Arc<DataSchema>,
+    plan1: Option<&CopyIntoTablePlan>,
+    plan2: Option<&DistributedCopyIntoTable>,
+    ctx: Arc<QueryContext>,
+    to_table: Arc<dyn Table>,
+    files: Vec<StageFileInfo>,
+    start: Instant,
+    use_commit: bool,
+) -> Result<()> {
+    let plan_required_source_schema: &DataSchemaRef;
+    let plan_required_values_schema: &DataSchemaRef;
+    let plan_values_consts: &Vec<Scalar>;
+    let plan_stage_table_info: &StageTableInfo;
+    let plan_force: &bool;
+    let plan_write_mode: &CopyIntoTableMode;
+
+    if let Some(plan) = plan1 {
+        plan_required_source_schema = &plan.required_source_schema;
+        plan_required_values_schema = &plan.required_values_schema;
+        plan_values_consts = &plan.values_consts;
+        plan_stage_table_info = &plan.stage_table_info;
+        plan_force = &plan.force;
+        plan_write_mode = &plan.write_mode;
+    } else if let Some(plan) = plan2 {
+        plan_required_source_schema = &plan.required_source_schema;
+        plan_required_values_schema = &plan.required_values_schema;
+        plan_values_consts = &plan.values_consts;
+        plan_stage_table_info = &plan.stage_table_info;
+        plan_force = &plan.force;
+        plan_write_mode = &plan.write_mode;
+    } else {
+        unreachable!();
```

Comment:
seems the plans passed in shall be of type `Option<(&CopyIntoTablePlan, &DistributedCopyIntoTable)>`?